addons/web/static/src/js/views/abstract_view.js

```diff
@@ -133,20 +133,19 @@ var AbstractView = Class.extend({
     getController: function (parent) {
         var self = this;
         return $.when(this._loadData(parent), this._loadLibs()).then(function () {
-            var model = self.getModel();
-            var state = model.get(arguments[0]);
+            var state = self.model.get(arguments[0]);
             var renderer = self.getRenderer(parent, state);
             var Controller = self.Controller || self.config.Controller;
             var controllerParams = _.extend({
                 initialState: state,
             }, self.controllerParams);
-            var controller = new Controller(parent, model, renderer, controllerParams);
+            var controller = new Controller(parent, self.model, renderer, controllerParams);
             renderer.setParent(controller);
 
-            if (!self.model) {
+            if (self.model.getParent() === parent) {
```

Comment:
instead of using this heuristic, i suggest to check at line 134:
```
// check if a model already exists, as if not, one will be created and
// we'll have to set the controller as its parent
var alreadyHasModel = !!this.model;
```
and do
`if (!alreadyHasModel) {`
here.

I think that this is equivalent, but it's more explicit.

Reply:
I'll try your solution